tests/src/core/testqgsgeometry.cpp

```diff
@@ -7826,6 +7828,215 @@ void TestQgsGeometry::circle()
   QGSCOMPARENEAR( l2p2.y(), -0.897, 0.01 );
 }
 
+void TestQgsGeometry::quadrilateral()
+{
+
+  // default
+  QgsQuadrilateral quad_init;
+  QgsPointSequence pts = quad_init.points();
+  QVERIFY( pts.at( 0 ) == QgsPoint() );
+  QVERIFY( pts.at( 1 ) == QgsPoint() );
+  QVERIFY( pts.at( 2 ) == QgsPoint() );
+  QVERIFY( pts.at( 3 ) == QgsPoint() );
+  QVERIFY( quad_init.isValid() == false );
+
+  // colinear
+  QgsQuadrilateral quad4points_col( QgsPoint( 0, 0 ), QgsPoint( 0, 5 ), QgsPoint( 0, 10 ), QgsPoint( 10, 10 ) );
+  QVERIFY( quad4points_col.isValid() == false );
+  pts = quad4points_col.points();
+  QVERIFY( pts.at( 0 ) == QgsPoint() );
```

Comment:
```suggestion
  QCOMPARE( pts.at( 0 ), QgsPoint() );
```

That way you get better debugging output if the test fails